tsconfig.json

```diff
@@ -0,0 +1,48 @@
+{
+  "compileOnSave": true,
+  "compilerOptions": {
+    "stripInternal": true,
+
+    // Enables all strict type checking options.
+    "strict": true,
+
+    // Library files to be included in the compilation. Basically which "core
+    // language features" TypeScript should enable.
+    "lib": ["es2015", "es2016", "es2017"],
+
+    // Which version of EcmaScript TypeScript should transpile to. Because of
+    // how Babel and Webpack is set up we currently target `esnext`, and let
+    // Babel take care of transpiling it down.
+    "target": "esnext",
+
+    // Specifies module code generation.
+    "module": "esnext",
+
+    // Allows default imports from modules with no default export. This does not affect code emit, just type checking.
+    "allowSyntheticDefaultImports": true,
+
+    // Emits __importStar and __importDefault helpers for runtime babel ecosystem compatibility.
+    "esModuleInterop": true,
+
+    // Resolve modules in the same way as Node.js. Aka make `require` works the
+    // same in TypeScript as it does in Node.js.
+    "moduleResolution": "node",
+
+    // Disallow inconsistently-cased references to the same file.
+    "forceConsistentCasingInFileNames": true,
+
+    // Report error when not all code paths in function return a value.
+    "noImplicitReturns": false,
+
+    // Report errors on unused locals.
+    "noUnusedLocals": true,
```

Comment:
I know this is `true` in `new-platform`, but I think we should turn it off and rather focus on improving the lint setup. This is annoying when just exploring things in code, but not commiting yet.

Reply:
I think we can also just remove the `noImplicitReturns` flag from here. (It's also `false` by default)

Reply:
Yeah it (`noUnusedLocals` and `noImplicitReturns`) feels like linter responsibility, my only minor concern is that we don't have linter setup yet. On the other hand it shouldn't do any harm, so I'll remove these settings.